### PR TITLE
Support PHP 8.4

### PR DESCRIPTION
**Pull request by lolli42**
Resolves: #1107

### CODE REVIEW

**Comment by lolli42** [comment]
Fixed using the two new rules in php-cs-fixer.

**Comment by lolli42** [comment]
Friendly ping.

**Review by crynobone — APPROVED** [review]

**Comment by lolli42** [comment]
Friendly ping.
We are at PHP 8.4.0 rc2 now. Would be great to have a compat release of this library.

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -77,6 +77,10 @@
         'no_whitespace_before_comma_in_array' => ['after_heredoc' => true],
         'no_whitespace_in_blank_line' => true,
         'non_printable_character' => true,
+        'nullable_type_declaration' => [
+            'syntax' => 'question_mark',
```
Comment (uuf6429) [inline .php-cs-fixer.dist.php:81]: Just happened to see this change.

I would suggest against this, it doesn't play well with union types. E.g. `?int|float` is invalid. Just a thought.

Comment (uuf6429) [reply .php-cs-fixer.dist.php:81]: Hmm, but on the other hand union types are not supported before 8.0, meanwhile 8.4 shows a deprecation message for implicit null types. 🙄

Guess this is the only way then.

**Review by OndraM — APPROVED** [review]
Nice, thank you!

**Comment by crynobone** [comment]
@OndraM Can we have this release ASAP (as PHP 8.4 has been released)

**Comment by OndraM** [comment]
Version 1.15.2 is now released.

**Pull request merged**